### ISSUE TITLE
"conan install" should use "conanfile.txt" in the current directory by default !

### ISSUE THREAD

**Issue by ghking1** [conan-io/conan#5322]
as for "pip install" and "npm install", no one enforce user to add a "point", the "point" is really ignorable for new users when they read documents. please remove it, thank you!

**Comment by memsharded** [comment]
Hi @ghking1 

It was that way in the past, but we decided to add it, for consistency. Note that most of Conan commands are:

```bash
$ conan command <path> <reference> [arguments]
```

So it is possible to do:
```bash
$ conan install . myuser/mychannel
```
To define that the conanfile.py at the current location is using ``myuser/mychannel`` as user and channel. That won't be possible if we remove the need to define the ``<path>``. But if the point is not necessary it would be weird to need it now.

So we prefer to keep the consistency and symmetry of the conan commands. Thanks for your feedback!

**Comment by ghking1** [comment]
Thanks for your reply @memsharded 

1. You can let path to be point defaultly when people just want to input "conan install", also "conan install . " should work, so it wouldn't affect old user‘s habit .

2. I'm try to use conan and artifactory in my group, But let everyone remember the "point" is not easy, especially for these people who are not easy to accept new things. I have been asked so much times about this problem these days, someone even think it slows down thieir work because of this new tool and angry at me. You should know, not every one is good at command line.

3. Does it another way of symmetry with other popular language, if it allow to use "conan install" ?

Of course, I respect your decision. If you don't want to make change, I may use some other not elegant way to reduce confusion for my colleague.

**Comment by memsharded** [comment]
Hi @ghking1 

I understand your pain, and it is something that I have experienced myself a few times. I remember when introducing git for the first time... so many upset devs...

The problem with hiding this dot is that it will bite you later. It is likely that you might soon need to specify user/channel, or even version@user/channel when you install. They are not infrequent. So if you are not used to type the dot, you will type:

```bash
$ conan install user/channel # Fail
```

And then the system will fail, because it is impossible to distinguish that from a path. And they will wonder again why the dot is necessary, but removing it is impossible, because that would be totally breaking. As I commented, we had it exactly that way, and we decided to remove it, because it causes less confusion to be consistent than the opposite. Because soon they might want to do:

```bash
$ conan inspect . # Summarizes info from the conanfile.py in this location
$ conan info .  # to see info from the dependency graph
$ conan build .  # To fire a build locally to debug something
etc
```
And the dot can't simply be removed from those commands.

If remembering to type a "." dot is causing some questions, I'd consider that just as a symptom of other issues, but that is not an issue by itself. Maybe you want to explore other approaches trying to automate things for them, like if you are using cmake, using the cmake-conan wrapper that calls conan automatically from cmake. Or maybe the CLion plugin if using CLion, or the recently released Visual Studio extension. Other users decide to put a "build.bat" or "build.sh" in the root of every repo, with the necesssary commands, there, so developers don't need to worry about commands, just execute that.
